src/server/game/Entities/AreaTrigger/AreaTriggerTemplate.h

```diff
@@ -71,6 +72,16 @@ enum AreaTriggerActionUserTypes
     AREATRIGGER_ACTION_USER_MAX    = 6
 };
 
+struct AreaTriggerServerPosition
+{
+    uint32 Id;
+    bool IsServer;
```

Comment:
You should replace these two with either direct pointer to AreaTriggerTemplate or encapsulate using AreaTriggerTemplateKey